src/main/java/io/cdap/plugin/switchcase/route/BasicPortSpecificationEvaluator.java

```diff
@@ -0,0 +1,151 @@
+/*
+ *  Copyright © 2020 Cask Data, Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ *  use this file except in compliance with the License. You may obtain a copy of
+ *  the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ *  WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ *  License for the specific language governing permissions and limitations under
+ *  the License.
+ */
+
+package io.cdap.plugin.switchcase.route;
+
+import com.google.common.base.Joiner;
+import com.google.common.base.Splitter;
+import com.google.common.base.Strings;
+import io.cdap.cdap.api.data.format.StructuredRecord;
+import io.cdap.cdap.etl.api.FailureCollector;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+/**
+ * A {@link PortSpecificationEvaluator} that evaluates port specifications in the basic routing mode
+ */
+public class BasicPortSpecificationEvaluator implements PortSpecificationEvaluator {
+  private static final Logger LOG = LoggerFactory.getLogger(BasicPortSpecificationEvaluator.class);
+
+  private final String routingField;
+  private final String nullPort;
+  private final List<BasicPortSpecification> portSpecifications;
+
+  BasicPortSpecificationEvaluator(String routingField, String nullPort, String portSpecification,
+                                  FailureCollector collector) {
+    this.routingField = routingField;
+    this.nullPort = nullPort;
+    this.portSpecifications = parse(portSpecification, collector);
+  }
+
+  @Override
+  public List<String> getAllPorts() {
+    List<String> ports = new ArrayList<>();
+    for (BasicPortSpecification specification : portSpecifications) {
+      ports.add(specification.getName());
+    }
+    return ports;
+  }
+
+  @Override
+  public String getPort(StructuredRecord record) throws PortNotSpecifiedException {
+    Object value = record.get(routingField);
+    // if the value is null, emit it based on the selected null handling option
+    if (value == null) {
+      LOG.trace("Found null value for {}.", routingField);
```

Comment:
This doesn't show in logs. Please use LOG.debug.

Reply:
fixed